### ISSUE TITLE
Unable to Enable Font Ligatures with Monaspace Font in IntelliJ IDEA 2023.2.2

### ISSUE THREAD

**Issue by zyfgisrs** [githubnext/monaspace#65]
Hello,

I am encountering an issue with enabling font ligatures in IntelliJ IDEA 2023.2.2 using the Monaspace font. Despite following the standard procedure for enabling ligatures in the IDE, they do not appear to be working.

### Issue Description
Environment: IntelliJ IDEA 2023.2.2
Font: Monaspace 
Issue: The ligatures feature of the Monaspace font is not functioning as expected.
### Steps to Reproduce
Open IntelliJ IDEA 2023.2.2 and navigate to File > Settings > Editor > Font.
Select the Monaspace font.
Enable the option "Enable font ligatures".
Apply the changes and observe that ligatures are not displayed in the editor.
### Expected Behavior
The expected behavior is that upon enabling the "Enable font ligatures" option for the Monaspace font, ligatures should be visible in the code editor, as they are in other IDEs like Visual Studio Code.

### Actual Behavior
Despite enabling the ligatures option, they are not being displayed in the IntelliJ IDEA editor when using the Monaspace font.

I would appreciate any assistance in resolving this issue or understanding if there are any additional steps or configurations required to enable font ligatures with Monaspace in IntelliJ IDEA 2023.2.2.

**Comment by Byron** [comment]
I thought it was interesting to share what happens when ligatures are not displayed, here are some pictures:

##### Without Ligatures

<img width="670" alt="without" src="https://github.com/githubnext/monaspace/assets/63622/03b52523-ec71-4b15-b971-ba0a7e9be9c7">

##### With Ligatures

<img width="696" alt="with" src="https://github.com/githubnext/monaspace/assets/63622/05291d6c-eb34-471a-860d-ac1fb031289a">

Wherever there would be a ligature, the letters disappear. For instances, `maximize` becomes `mxze`.

**Comment by ishenyi** [comment]
Same on  IntelliJ IDEA 2023.2.5

**Comment by fanf** [comment]
Same problem, IDEA 2023.2.4 and 2023.2.5. 
System is archlinux with fontconfig 2.14.2 if it may matter.

**Comment by dantereve** [comment]
The thing is that IntelliJ-based IDE does not support stylistic sets: https://youtrack.jetbrains.com/issue/IDEA-221886/Support-stylistic-sets-for-fonts . 

All ligatures seem tied to a specific stylistic set in those fonts, which is not the case for many coding fonts. I guess they had to do it this way.

**Comment by xonev** [comment]
I'm on 2023.2.1 on macOS 14.1 and it seems to work great for me:

<img width="744" alt="Screenshot 2023-11-14 at 15 41 11" src="https://github.com/githubnext/monaspace/assets/142312/67930a47-e095-4dea-bbb5-5b72b5d76c44">
<img width="731" alt="Screenshot 2023-11-14 at 15 41 21" src="https://github.com/githubnext/monaspace/assets/142312/51f784ec-3237-477f-8910-dae71e444232">

**Comment by vonhyou** [comment]
> I'm on 2023.2.1 on macOS 14.1 and it seems to work great for me:
> 
> <img alt="Screenshot 2023-11-14 at 15 41 11" width="744" src="https://user-images.githubusercontent.com/142312/282962587-67930a47-e095-4dea-bbb5-5b72b5d76c44.png"> <img alt="Screenshot 2023-11-14 at 15 41 21" width="731" src="https://user-images.githubusercontent.com/142312/282962597-51f784ec-3237-477f-8910-dae71e444232.png">

Pls check the last two lines of your second screenshot, apparently it's not working.

![](https://user-images.githubusercontent.com/22723/281849352-49b4f802-265d-414c-94c0-ec712e3c0ecc.png)
![image](https://github.com/githubnext/monaspace/assets/29454668/2c97467a-6c78-46a5-bddd-63a440a6c255)

![image](https://github.com/githubnext/monaspace/assets/29454668/d06e92b5-37a3-476b-a009-3ec9e6ef64d0)

![image](https://github.com/githubnext/monaspace/assets/29454668/93313d26-9ed7-4bc2-9927-db35b2da1331)

**Comment by xonev** [comment]
Excuse me, you're right that the actual ligatures aren't working but the texture healing does work correctly when I enable ligatures. That was broken as well for the OP.

**Comment by savva312** [comment]
Hello, i am experiencing the same behavior with WebStorm 2023.2.5. Any news/updates?

**Comment by josh-degraw** [comment]
Yeah this feels frustrating that I _have to_ individually opt-in to enable ligatures, which I can't do in Rider. I just want all ligatures on always if I set "enableFontLigatures" in any editor. Is there not a way to do this with how the fonts are created?

**Comment by EvanNotFound** [comment]
Same situation here with Webstorm 2023.2.5. Any updates?

**Comment by maragnus** [comment]
Both Intellij and Rider 2023.3.1 do not show Monaspace ligatures for me. But ligatures work on both with the JetBrains Mono font.

**Comment by ai212983** [comment]
Guys, please read the thread carefully. Stylistic sets (i.e. custom ligatures) are [NOT supported](https://youtrack.jetbrains.com/issue/IDEA-221886/Support-stylistic-sets-for-fonts) in IntelliJ products yet. 

One can use [OpenType Feature Freezer](https://twardoch.github.io/fonttools-opentype-feature-freezer/) to generate fonts for IntelliJ at his own risk or wait for [OpenType features freezing ticket](https://github.com/githubnext/monaspace/issues/162) to be implemented.

**Comment by josh-degraw** [comment]
I for one am well aware that the IDEA issue is on their side. I'm specifically asking if in this font library there can be a subset of fonts generated with all ligatures enabled that doesn't require manually configuring OTT features for editors that don't yet support it. I believe there are other fonts that do this for this purpose until these OTT features have wider support.

**Comment by ai212983** [comment]
@josh-degraw, I believe [the ticket I mentioned](https://github.com/githubnext/monaspace/issues/162) should cover that.

As a side note, Monaspace uses font variables (`width`, `slant`, fine-grained `weight`) which most IDEs can't configure. For instance, after using the narrower [PragmataPro](https://fsd.it/shop/fonts/pragmatapro/), Monaspace in IDEA appears overly wide.

Using the [fontTools library ](https://github.com/fonttools/fonttools) for batch converting variable Monaspace fonts could work as a solution until **@githubnext** adds the feature/variable freezing.

**Comment by VsedanoVargas** [comment]
¿Qué, a ustedes les aparece para elegir la fuente monaspace? pensé que esa tipografía no era compatible con intellij idea. tengo la versión "IntelliJ IDEA 2024.1.4 (Ultimate Edition)" y para hoy 22 de julio de 2024 no aparece ninguna fuente monaspace en las opciones tipográficas del IDE.

![image](https://github.com/user-attachments/assets/7c06105f-4f26-4560-a63e-b315cd8d1e06)
![image](https://github.com/user-attachments/assets/cc6cbf20-c79f-4c29-80f2-2ea055048893)

**Comment by idan** [comment]
The new Frozen fonts in 1.2 are designed to solve this issue for environments which do not let you control OpenType features properly, like IntelliJ and XCode.

We'd prefer that those editors properly support typographical features, but we hate for you to use other fonts. So we made the frozen fonts. Just for you. 💖
